hibernate-core/src/main/java/org/hibernate/engine/jdbc/connections/internal/DriverManagerConnectionProviderImpl.java

```diff
@@ -453,4 +449,17 @@ public void closeConnection(Connection conn) throws SQLException {
 		}
 	}
 
+	private static class ValidationThreadFactory implements ThreadFactory {
+
+		public static final ValidationThreadFactory INSTANCE = new ValidationThreadFactory();
```

Comment:
Is it really a good idea? I mean, most of the time, we will use it only once and we'd better get it garbage collected rather than kept as a static thing forever?

Reply:
If you finally decide to keep the INSTANCE variable, taking into account the scope of the class it can be enough to define it as private, at least for the moment

Reply:
It's better to remove it. The only time the ConnectionProvider will be stopped is when the SF is closed, so this should be a one-time instance creation.